File: day3/c/main.c
#include <stdio.h>
#include <stdbool.h>

void main() {
	// Char / String
	char value1 = 't';
	char value2[] = "test"; // Dynamic

	int value3 = 1; // 4 bytes
	short value4 = 1; // 2 bytes
	long value5 = 1; // 4 / 8 bytes
	long long value6 = 30; // 8 bytes

	// unsigned -> Does not allow the variable to have negative numbers

	float value7 = 10.00; // 4 bytes
	double value8 = 10.00; // 8 bytes
	long double value9 = 10.00; // 8/12/16 bytes

	bool value10 = true;

	int value11[5] = {1, 2, 3, 4, 5}; // Fixed size array

	char *value12 = &value1; // Pointer

	printf("%c\n", value1);
	printf("%s\n", value2);
	printf("%d\n", value3);
	printf("%d\n", value4);
	printf("%ld\n", value5);
	printf("%lld\n", value6);
	printf("%f\n", value7);
	printf("%.2f\n", value7); // Limit float return
	printf("%f\n", value8);
	printf("%Lf\n", value9);
	printf("%d\n", value10);
	printf("%d\n", value11[0]);
	printf("%c\n", *value12);
}

File: day3/go/main.go
package main

import "fmt"

// Custom type
type TestValue struct {
	t1 string
	t2 string
}

func main() {
	var value1 int = 10
	var value2 string = "t1"
	var value3 bool = true
	var value4 float32 = 1.1
	var value5 float64 = 1.1
	var value6 *int = &value1 // Memory reference
	value7 := 1               //Inference -> The language engine evaluates the data and has the type at the time of assembly

	var v1, v2, v3 int = 1, 2, 3

	// Array
	var value8 = [2]int{1, 2}
	value9 := [2]int{1, 2}

	value10 := TestValue{"t1", "t2"}

	// Constants
	const value11 int = 10

	fmt.Println(value1)
	fmt.Println(value2)
	fmt.Println(value3)
	fmt.Println(value4)
	fmt.Println(value5)
	fmt.Println(value6)
	fmt.Println(value7)
	fmt.Println(v1, v2, v3)
	fmt.Println(value8)
	fmt.Println(value9)
	fmt.Println(value10)
	fmt.Println(value11)
}
